parlai/mutators/flatten.py

```diff
@@ -23,4 +23,5 @@ def many_episode_mutation(self, episode: List[Message]) -> List[List[Message]]:
         for message in episode:
             history.append(message.pop('text'))
             message['text'] = '\n'.join(history)
+            history.extend(message['labels'])
```

Comment:
this should be after the yield, and it shouldn't extend with all of labels. It should pick a random element from labels.

Reply:
I can change it to pick one random element from the labels. 
Why putting it after yield matters? We have already used history and are not going to use it again until next iterator call. Does it have something to do with the another thread that possibly calls the function?

Reply:
You're leaking the label!

Reply:
Discussed offline why the implementation is correct, but potentially confusing.